lib/globus_client/endpoint.rb

```diff
@@ -199,7 +199,8 @@ def access_rule
 
       # debugging of Globus responses
       response_body = JSON.parse(response.body)
-      raise(StandardError, "Response is missing DATA in: #{response_body}") unless response_body.key?("DATA")
+
+      UnexpectedResponse.call(response, message: "Response is missing DATA in: #{response_body}") unless response.success? && response_body.key?("DATA")
```

Comment:
why call instead of raise?

Reply:
That class has a method called `#call` which examines the response and then raises different exceptions based on what is in the response.  It's a blanket wrapper for a bunch of logic about how to handle the different responses that come back.

Reply:
see https://github.com/sul-dlss/globus_client/blob/main/lib/globus_client/unexpected_response.rb#L29-L45

Reply:
If we raise `StandardError` here, we short-circuit the ability to refresh the token on a 401. Using `UnexpectedResponse.call` will raise a more targeted custom exception, e.g., `UnauthorizedError`, which is what the `TokenWrapper` uses to detect the situation where a token needs to be refreshed.